docs/reference/contributing/connectivity/ethernet.md

```diff
@@ -0,0 +1,132 @@
+
+### Wi-Fi interfaces
+
+As a Wi-Fi interface, a little more work is necessary - at a minimum, you need to implement the extra configuration calls in `WiFiInterface`. This is because the network stacks and EMAC APIs only relate to the Ethernet-like data path - they have no knowledge of any other configuration mechanisms and assume they are already set up.
+
+To do this, create a C++ class that inherits from both `WiFiInterface` and `EMACInterface`. The `EMACInterface` is a helper class that implements all the core `NetworkInterface` functionality for you. You then need to implement the extra `WiFiInterface` configuration methods.
+
+For reference, `EthernetInterface` also derives from `EMACInterface` but has no extra code because no extra configuration is necessary.
+
+You don't usually directly expose the `EMAC` class of a Wi-fi driver - it is not usually declared as `EMAC::get_default_instance`, but you would pass it to the constructor of your base `EMACInterface`. This then makes it visible using the `get_emac` method. This is for test purposes, meaning the test framework can do:
+
+```
+    MyWiFiInterface net;
+    net.set_credentials();
+    EMAC &emac = net.get_emac();
+    do_emac_test(emac);
+```
+
+This must work in your driver - it must be possible to power up and use the built-in EMAC directly without the `NetworkInterface::connect()` method being invoked, as long as the credentials have been set. This structure comes naturally if you use the default `EMACInterface::connect()` implementation.
+
+Note also that your constructor must allow the network stack to be specified using the same form as `EthernetInterface`:
+
+```
+    MyWiFiInterface(OnboardNetworkStack &stack = OnboardNetworkStack::get_default_instance());
+```
+
+### OnboardNetworkStack
+
+The precise details of the `OnboardNetworkStack` API should not concern an EMAC driver writer - it provides the mechanism to bind a driver to a stack, and the APIs needed to implement a `NetworkInterface`, but this is handled by `EMACInterface`, either as a base class of your own `XXXInterface` or as the base of `EthernetInterface`.
+
+### DEVICE_EMAC
+
+At present, as an interim measure, targets providing `EMAC::get_default_instance()` should add "EMAC" in `device_has` in their `targets.json`. This activates network tests in CI builds.
+
+This is subject to change, but is necessary in lieu of the previous typical behavior of gating tests on `FEATURE_LWIP`.
+
+### Tuning memory allocations
+
+Depending on its use of pool and heap memory and other factors, a driver might want to tune the configuration of particular network stacks. You can do this using the `mbed_lib.json` of each network stack, using their `target_overrides` section.
+
+### Testing
+
+The Mbed OS tree contains Greentea-based tests that exercise the EMAC API directly, and more general socket tests.
+
+See here for general Greentea information: <https://github.com/ARMmbed/greentea>
+
+See here for the EMAC tests:
```

Comment:
We should link to these in Doxy instead of in GH.

Reply:
Greentea tests are not documented anywhere.
There is no Doxygen or Handbook page about those.